doc/subsystems/jinjagen/jinjagen.rst

```diff
@@ -0,0 +1,256 @@
+..
+    Copyright (c) 2004-2015 Ned Batchelder
+    SPDX-License-Identifier: MIT
+    Copyright (c) 2018 Bobby Noelte
+    SPDX-License-Identifier: Apache-2.0
+    Copyright (c) 2018 Nordic Semiconductor ASA
+    SPDX-License-Identifier: Apache-2.0
+.. _jinjagen_intro:
+
+Introduction
+############
+
+For some repetitive or parameterized coding tasks, it's convenient to
+use a code generating tool to build C code fragments, instead of writing
+(or editing) that source code by hand. Such a tool can also access CMake build
+parameters and device tree information to generate source code automatically
+tailored and tuned to a specific project configuration.
+
+The Zephyr project supports a code generating tool that processes template files
+and generates output files. It can be used, for example, to generate source code 
+that creates and fills data structures, adapts programming logic, creates
+configuration-specific code fragments, and more.
+
+.. image:: template_flow.png
+   :width: 500px
+   :align: center
+   :alt: Principle
+
+The code generator uses a template engine for python called jinja2. The jinja2
+templates are typically stored in files with the file extention .jinja2 which are
+rendered into other files like c-source files. The documentation for jinja2 can 
+be found here `Jinja2 <http://jinja.pocoo.org/>`_. Please refer to this site for more
+extensive information, the information in the following sections are short
+descriptions of some of the key features in jinja2. 
+
+Jinja2 files are rendered into other files using the cmake extensions functions
+
+  * zephyr_sources_jinjagen(..)
```

Comment:
renders better if you don't indent these lines (if you view this file in GitHub you'll see what I mean).  With too much indentation like this, the renderer adds a vertical bar on the left thinking it's a blockquote.

Reply:
ok